cirq/circuits/circuit.py

```diff
@@ -1314,6 +1324,49 @@ def _draw_moment_in_diagram(moment: Moment,
         if exponent is not None:
             out_diagram.write(x, y2, '^' + exponent)
 
+    # Group together columns belonging to the same Moment.
+    if moment.operations and x > x0:
+        moment_groups.append([x0, x])
+
+
+def _draw_moment_groups_in_diagram(moment_groups: List[Tuple[int, int]],
+                                   use_unicode_characters: bool,
+                                   out_diagram: TextDiagramDrawer,
+                                   transpose: bool):
+    out_diagram.shift_vertically(0, 1)
+    h = out_diagram.height()
+
+    left_end = '├' if use_unicode_characters else '|'
+    top_end = '┬' if use_unicode_characters else '-'
+    right_end = '┤' if use_unicode_characters else '|'
+    bottom_end = '┴' if use_unicode_characters else '-'
+
+    # Insert columns starting from the back since the insertion
+    # affects subsequent indices.
+    for (x1, x2) in reversed(moment_groups):
+        out_diagram.shift_horizontally(x2 + 1, 1)
+        out_diagram.force_horizontal_padding_after(x2, 0)
+
+        for y in [0, h]:
+            out_diagram.write(x2 + 1, y, right_end, bottom_end)
+        out_diagram.force_horizontal_padding_after(x2 + 1,
+            2 if not transpose else 0)
```

Comment:
Why does this depend on being transposed or not?

Reply:
This is needed to match the spacing logic in `to_text_diagram`, which sets `horizontal_spacing=1 if transpose else 3`.